backend/dataall/core/environment/services/managed_iam_policies.py

```diff
@@ -0,0 +1,156 @@
+from typing import List
+import logging
+import json
+from abc import ABC, abstractmethod
+from dataall.base.aws.iam import IAM
+
+logger = logging.getLogger(__name__)
+
+
+class ManagedPolicy(ABC):
+    """
+    Abstract class for any IAM Managed policy that needs to be atached to data.all team IAM roles and consumption roles
+    """
+
+    @abstractmethod
+    def __init__(
+            self,
+            role_name,
+            account,
+            environmentUri,
+            resource_prefix
+    ):
+        self.role_name = role_name
+        self.account = account
+        self.environmentUri = environmentUri
+        self.resource_prefix = resource_prefix
+
+    @property
+    @abstractmethod
+    def policy_type(self):
+        """
+        Returns string and needs to be implemented in the ManagedPolicies inherited classes
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def generate_policy_name(self) -> str:
+        """
+        Returns string and needs to be implemented in the ManagedPolicies inherited classes
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def generate_empty_policy(self) -> dict:
+        """
+        Returns dict and needs to be implemented in the ManagedPolicies inherited classes
+        """
+        raise NotImplementedError
+
+    def check_if_policy_exists(self) -> bool:
+        policy_name = self.generate_policy_name()
+        share_policy = IAM.get_policy_by_name(self.account, policy_name)
+        return (share_policy is not None)
+
+    def check_if_policy_attached(self):
+        policy_name = self.generate_policy_name()
+        return IAM.is_policy_attached(self.account, policy_name, self.role_name)
+
+    def attach_policy(self):
+        policy_arn = f"arn:aws:iam::{self.account}:policy/{self.generate_policy_name()}"
+        try:
+            IAM.attach_role_policy(
```

Comment:
does `attach_role_policy()` ever raise an exception?

Reply:
In this [commit](https://github.com/data-dot-all/dataall/pull/1068/commits/65cf21a507086e3587ecfa0e8e22d52e8bf7601a) I added error handling to be propagated up to the API. I also restructured the resolvers a bit so that they will fail if they cannot create the policies before creating the RDS rows